src/sentry/integrations/jira/integration.py

```diff
@@ -716,20 +717,20 @@ def sync_assignee_outbound(self, external_issue, user, assign=True, **kwargs):
         if assign:
             for ue in user.emails.filter(is_verified=True):
                 try:
-                    res = client.search_users_for_issue(external_issue.key, ue.email)
+                    possible_users = client.search_users_for_issue(external_issue.key, ue.email)
                 except (ApiUnauthorized, ApiError):
                     continue
-                try:
-                    jira_user = [
-                        r
-                        for r in res
-                        if r.get("emailAddress") and r["emailAddress"].lower() == ue.email.lower()
-                    ][0]
-                except IndexError:
-                    pass
-                else:
-                    break
-
+                for possible_user in possible_users:
+                    email = possible_user.get("emailAddress")
+                    # pull email from API if we can use it
+                    if email is None and settings.JIRA_USE_EMAIL_SCOPE:
```

Comment:
This path is also used for jira-server. While we shouldn't run into GDPR email redaction there the availability of the v3 emails endpoint is likely going to be poor and we might want to capture request failures here to not break sync should a user have an odd shaped configuration.

Reply:
@markstory Good call out, I should make sure we don't do this with Jira server.

Reply:
@markstory Added some code so jira_server won't try to use the external email api